nan.h

```diff
@@ -90,12 +90,223 @@
 #define IOJS_1_0_MODULE_VERSION  42
 #define IOJS_1_1_MODULE_VERSION  43
 
+//=== CallbackInfo =============================================================
+
+#if NODE_MODULE_VERSION > NODE_0_10_MODULE_VERSION
+typedef v8::FunctionCallbackInfo NanFunctionCallbackInfo;
+typedef void(*NanFunctionCallback)(const NanFunctionCallbackInfo<v8::Value>&);
+
+typedef v8::PropertyCallbackInfo NanPropertyCallbackInfo;
+typedef void(*NanGetterCallback)(const NanPropertyCallbackInfo<v8::Value>&);
+typedef void(*NanSetterCallback)
+    (const NanPropertyCallbackInfo<v8::Value>&, v8::Handle<v8::Value>);
+#else
+
+namespace Nan { namespace imp {
+  template<typename T>
+  class NanReturnValueImp;
+}
+}
+
+template<typename T>
+class NanReturnValue {
+  v8::Handle<v8::Value> value_;
+  friend class Nan::imp::NanReturnValueImp<v8::Value>;
+
+ public:
+  inline NanReturnValue() {}
+  template <class S> inline NanReturnValue(const NanReturnValue<S>& that)
+      : value_(that.value_) {
+    *(static_cast<T *volatile*>(0)) = static_cast<S*>(0);
+  }
+
+  v8::Handle<v8::Value> Value() { /*assert(!value_.IsEmpty());*/ return value_; }
+
+  // Handle setters
+  template <typename S> inline void Set(v8::Handle<S> handle) {
+    fprintf(stderr, "retval setter!!\n");
+    value_ = handle;
+    assert(!value_.IsEmpty());
+    fprintf(stderr, "Retval successfully set to %s\n", *v8::String::Utf8Value(value_.As<v8::String>()));
+    fprintf(stderr, "end retval setter!!\n");
+  }
+
+  // Fast primitive setters
+  inline void Set(bool value) { value_ = v8::Boolean::New(value); }
+  inline void Set(double i) { value_ = v8::Number::New(i); }
+  inline void Set(int32_t i) { value_ = v8::Int32::New(i); }
+  inline void Set(uint32_t i) { value_ = v8::Uint32::New(i); }
+
+  // Fast JS primitive setters
+  inline void SetNull() { value_ = v8::Null(); }
+  inline void SetUndefined() { value_ = v8::Undefined(); }
+  inline void SetEmptyString() { value_ = v8::String::Empty(); }
+};
```

Comment:
Yes, that would be necessary. I tried with `Persistent` earlier when trying to track down the problem at hand, but it was not the cause. Couldn't compile a debug build of Node 0.10.38 either, linking mksnapshot failed for unknown reason.

Another problem is deleting the function pointer wrapper class after running the callback body. Is v8 somehow taking control of the pointer I set with `SetPointerInInternalField`? Tried with `v8::External` earlier, but that was definitely managed by v8 and needed weak callbacks, which could be quite detrimental to performance.